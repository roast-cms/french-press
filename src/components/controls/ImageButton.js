import React from "react"
import styled from "styled-components"

import button from "./css/button"

const ImageButton = styled.button`
  ${button}
  position: absolute;
  border-radius: 0.25em;
  right: 0.5em;
`

/**
 * A button component that appears in the editor when user's carriage is on a new empty line. Have a good read through it if you'd like to build your own (you can plug it in to `<FrenchPress />` component).
 * @module ImageButton
 * @param {Boolean} followComposerCursor
 * @param {Function} onMouseDown
 */
export default props => {
  return (
    <ImageButton
      className="fpe-image-button"
      followComposerCursor
      style={props.style}
      onMouseDown={event => props.click(event)}
    >
      {props.children}
    </ImageButton>
  )
}
