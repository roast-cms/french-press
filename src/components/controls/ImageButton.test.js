import "jest-styled-components"

import React from "react"

import {shallow} from "enzyme"
import ImageButton from "./ImageButton"

test("Render ImageButton without crashing,  matches snapshot", () => {
  const element = shallow(<ImageButton />)

  expect(element).toMatchSnapshot()
})
