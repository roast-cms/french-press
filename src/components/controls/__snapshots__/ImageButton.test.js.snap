// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "className": "fpe-image-button",
      "followComposerCursor": true,
      "onMouseDown": [Function],
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  ",
          "
  outline: 0;
  background: ",
          [Function],
          ";
  border: none;
  color: ",
          [Function],
          ";
  font-size: 1em;
  padding: 0.05em 0.5em;
  font-family: inherit;
  cursor: pointer;
  &.active,
  &:active {
    background: ",
          [Function],
          ";
  }
",
          "
  position: absolute;
  border-radius: 0.25em;
  right: 0.5em;
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "className": "fpe-image-button",
        "followComposerCursor": true,
        "onMouseDown": [Function],
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  ",
            "
  outline: 0;
  background: ",
            [Function],
            ";
  border: none;
  color: ",
            [Function],
            ";
  font-size: 1em;
  padding: 0.05em 0.5em;
  font-family: inherit;
  cursor: pointer;
  &.active,
  &:active {
    background: ",
            [Function],
            ";
  }
",
            "
  position: absolute;
  border-radius: 0.25em;
  right: 0.5em;
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "usesTheme": true,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
